sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/binaryfile/BinaryFileFormatSuite.scala

```diff
@@ -352,15 +352,24 @@ class BinaryFileFormatSuite extends QueryTest with SharedSparkSession {
           .select(CONTENT)
       }
       val expected = Seq(Row(content))
-      QueryTest.checkAnswer(readContent(), expected)
+      QueryTest.checkAnswer(readContent(), expected) match {
+        case Some(errorMessage) => fail(errorMessage)
+        case None =>
+      }
       withSQLConf(SOURCES_BINARY_FILE_MAX_LENGTH.key -> content.length.toString) {
-        QueryTest.checkAnswer(readContent(), expected)
+        QueryTest.checkAnswer(readContent(), expected) match {
+          case Some(errorMessage) => fail(errorMessage)
+          case None =>
+        }
       }
       // Disable read. If the implementation attempts to read, the exception would be different.
       file.setReadable(false)
       val caught = intercept[SparkException] {
         withSQLConf(SOURCES_BINARY_FILE_MAX_LENGTH.key -> (content.length - 1).toString) {
-          QueryTest.checkAnswer(readContent(), expected)
+          QueryTest.checkAnswer(readContent(), expected) match {
```

Comment:
Can we .. 

1.  make `QueryTest.checkAnswer` private so that it can only be called in its companion class `QueryTest`
2. and avoid to call `checkAnswer` in the `QueryTest` object in the tests

?

Reply:
Ah, basically similar comment with @HeartSaVioR's.

Reply:
I think @HyukjinKwon's suggestion would be better for future modification in point of preventing the cause, if there's no actual usage of `QueryTest.checkAnswer` from outside of `QueryTest`. (I guess it might have been misused like this PR fixes...)